src/sentry_core.c

```diff
@@ -1169,3 +1169,29 @@ sentry_clear_crashed_last_run(void)
     sentry__options_unlock();
     return success ? 0 : 1;
 }
+
+void sentry_capture_minidump(sentry_path_t *dump_path, sentry_value_t event,
+    const sentry_options_t *options)
+{
+    sentry_envelope_t *envelope = sentry__prepare_event(
+        options, event, NULL, !options->on_crash_func);
+    sentry_session_t *session = sentry__end_current_session_with_status(
+        SENTRY_SESSION_STATUS_CRASHED);
+    sentry__envelope_add_session(envelope, session);
```

Comment:
The question is whether these two should run unconditionally or at all.

* does it make sense for this use case (or any other that essentially bypasses the SDK) to have a session or scope applied, given that this interface probably transfers an event from an entirely different session/scope?
* if we use `sentry__prepare_event()` here, its fourth parameter, `invoke_before_send`, should probably not depend on the `options->on_crash_func` being set (another reason not to extract the function and instead have two different implementations): even if we invoke a hook for an event that is entirely in the user's hands, it should always be `before_send` and not `on_crash` (which should strictly be called from a crash handler).
* if we don't apply scope or session here, then there are still a few things that `sentry__prepare_event()` does that might be useful:
  * checking for a valid `event_id` (`sentry__ensure_event_id()`)
  * creating an envelope to hold the event and minidump attachment
  * adding attachment items for the attachments configured via options (which could make sense?).
* as mentioned in https://github.com/getsentry/sentry-native/issues/1050#issuecomment-2404728084, while I think an event parameter would make this more generically useful, the question is what this parameter is going to be used for vs. auto-generating an event (which is what I had in mind when the argument is a null object).